### ISSUE TITLE
ADC Clock to set to fast for none T3.5/T3.6 devices.

### ISSUE THREAD

**Issue by Donziboy2** [PaulStoffregen/cores#240]
Looking at the Analog.c file I noticed that several of the 8 bit speed settings are above(18MHz) what a T3.2 and below boards are rated to.  
Is this a mistake or is it safe to run them at these speeds?

![adc clock](https://cloud.githubusercontent.com/assets/5633038/25305576/d8585bfe-274b-11e7-9eef-47df6870b96c.JPG)

**Comment by PaulStoffregen** [comment]
good question....

**Comment by pedvide** [comment]
The datasheet doesn't explicitly say that this can cause damage, only that the noise and resolution can't be guaranteed. 
Perhaps we could run some tests, maybe check the noise and resolution of a T3.2 at normal speed for a day or so, then repeat it for the the fastest speed, and check again at normal speeds. If the results at normal speeds are the same before and after then it should be fine.

**Comment by Donziboy2** [comment]
pedvide, I also compared your clocks.  See below your pushing the speeds a little ;)
![pedvide adc clk](https://user-images.githubusercontent.com/5633038/31766682-4df3950e-b496-11e7-863a-798b178480d6.JPG)

**Comment by pedvide** [comment]
I already fixed those errors, thanks a lot! (I haven't added the missing values yet though.)
If you use "very high speed" then indeed it can be above the maximum rating, but if you use "high speed" then it's always within specs.

Also notice that the maximum speed for T LC, 3.0, 3.1/3.2, and 3.5 is 18 MHz (below 16 bits), but for T3.6 it's 24 MHz.

**Comment by Donziboy2** [comment]
Cool, I will take a look and update my table.

I guess in the end its tough to say if it will just degrade the signal or cause actual damage.  I found out the hard way that using Pin13 as SCK while using DMA ADC was a disaster waiting to happen.  The trace for pin 13 on the T3.6 runs right under the micro from 1 side to the other, even with a 100R resistor in series it was adding 100mV pk-pk spikes onto my ADC readings.  Even when using alternate SCK pins it still looks awful(50mV pk-pk), I believe in the spec they say specs are accurate with other peripherals turned off/low power and I can see why.

**Comment by pedvide** [comment]
Yes. Critically, the specs are only given for differential 16 bit mode. I assume that the traces of the differential pins follow parallel courses inside the chip, so that all those perturbations cancel out.
